src/VirtoCommerce.Platform.Core/ObjectValue/ObjectValueJsonConverter.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Globalization;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace VirtoCommerce.Platform.Core.ObjectValue
+{
+    public class ObjectValueJsonConverter : JsonConverter
+    {
+        public override bool CanWrite => true;
+        public override bool CanRead => false;
+
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType.GetCustomAttributes(true).Any(a => a.GetType() == typeof(ObjectValueSerializedAsStringForIndexAttribute));
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var obj = JObject.FromObject(value, new JsonSerializer() { ContractResolver = serializer.ContractResolver });
+
+            if (value != null
+                && obj.TryGetValue("value", out var objValue)
+                && obj.TryGetValue("valueType", out var objValueType))
```

Comment:
Maybe we could determine value type based on the "deserialized" value, without dependency on `valueType` property